controller/internal/supervisor/supervisor.go

```diff
@@ -219,24 +219,30 @@ func (s *Config) doCreatePU(contextID string, pu *policy.PUInfo) error {
 func (s *Config) doUpdatePU(contextID string, pu *policy.PUInfo) error {
 
 	s.Lock()
+	defer s.Unlock()
 
-	data, err := s.versionTracker.LockedModify(contextID, revert, 1)
+	data, err := s.versionTracker.Get(contextID)
 	if err != nil {
 		return fmt.Errorf("unable to find pu %s in cache: %s", contextID, err)
 	}
 
 	c := data.(*cacheData)
-	if err := s.impl.UpdateRules(c.version, contextID, pu, c.containerInfo); err != nil {
+
+	if err := s.impl.UpdateRules(c.version^1, contextID, pu, c.containerInfo); err != nil {
 		// Try to clean up, even though this is fatal and it will most likely fail
-		s.Unlock()
+		zap.L().Error("Update rules failed with error", zap.Error(err))
```

Comment:
BTw, since we unsupervise at this point, the  value of the version don't make much different any more. it can be whatever.

Reply:
Yes, but Unsupervise calls DeleteRules which requires the correct version to be passed for it to delete the rules